src/main/scala/Joint.scala

```diff
@@ -62,9 +131,13 @@ case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Doub
     * vector for the plane, and the second item is the distance of the plane from the origin.
     */
   def globalCoordinates: Seq[((Double, Double, Double), Double)] = {
-    val Nplane = DenseVector[Double](a, b, c)
-    val strike = (dipDirection - math.Pi / 2) % (2 * math.Pi) // CHECKED: strike = dipDirection - pi/2 (US convention)
-    val Nstrike = DenseVector[Double](math.cos(strike), math.sin(strike), 0.0)
+    var normalDirection = -1.0 // Negative needed simply because of how vertical planes are dealt with
+    if (math.abs(c) > Joint.EPSILON) { // For non-vertical planes
+      normalDirection = -c/math.abs(c) // Ensures that normal will always point in -z global direction (ensures right-handed local coordinates)
+    }
+    val Nplane = normalDirection * DenseVector[Double](a, b, c)
+    val strike = (dipDirection - math.Pi / 2) % (2 * math.Pi) // Strike = dipDirection - pi/2 (US convention)
+    val Nstrike = DenseVector[Double](math.cos(- strike), math.sin(- strike), 0.0)
```

Comment:
Is there a reason not to write `-strike` instead of `- strike`? Usually people use the trick of avoiding spaces with the unary minus operator so that it looks different from binary subtraction.